### ISSUE TITLE
javacpp when compiles C++ files includes JDK paths which are not same as system Java

### ISSUE THREAD

**Issue by krzysg** [bytedeco/javacpp#398]
This is not a big problem but since I thought that I did something wrong (like wrong env settings etc.) it is worth to mention this issue for the others that might encounter same problem.

I have multiple java JDKs installed on my system (Ubuntu/Linux) and currently I am using Java 11:
```bash
$ echo $JAVA_HOME 
/usr/lib/jvm/jdk-11.0.7/

$ mvn -v
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 11.0.7, vendor: Oracle Corporation, runtime: /usr/lib/jvm/jdk-11.0.7
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-29-generic", arch: "amd64", family: "unix"

$ java -version
java version "11.0.7" 2020-04-14 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.7+8-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.7+8-LTS, mixed mode)
```

I have noticed that paths used by javacpp when compiling c++ srouce files are set for different JDK (jdk-14.0.1): 
```java
[INFO] g++ -I/home/gonciarz/ws/repo/JavaCppTemplate/src/main/cpp/pl/krzysg/ -I/usr/lib/jvm/jdk-14.0.1/include -I/usr/lib/jvm/jdk-14.0.1/include/linux /home/gonciarz/ws/repo/JavaCppTemplate/target/classes/pl/krzysg/cppLib/target/jniMyCppLibConfig.cpp /home/gonciarz/ws/repo/JavaCppTemplate/target/classes/jnijavacpp.cpp -march=x86-64 -m64 -O3 -s -std=c++14 -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -shared -o libjniMyCppLibConfig.so
```

It seems that when searching ```jni.h``` nad ```jni_md.h``` default/system java paths might not be used if there are other java JDK installed. The search is performed by this method:

https://github.com/bytedeco/javacpp/blob/3189a117080ca63ad93c6427cd367ed7395d5103/src/main/java/org/bytedeco/javacpp/tools/Builder.java#L116

Maybe it would be good to choose ```jni.h``` and ```jni_md.h``` from system java if possible?  
If this does not really matter this issue can be closed.

I have observed this for javacpp version 1.4.1 and 1.5.2 but since this code is very old it should behave same in every version from last years.

**Comment by saudet** [comment]
Java doesn't provide an API to query the location of these files, that's
why we need to search for them. What's happening here is probably because
for some reason your installation of JDK 11 is missing these files, but
luckily it finds them in your installation of JDK 14. Am I correct?

**Comment by krzysg** [comment]
Actually all my installed JDK's have needed files:

```
$ find ./ -name jni.h
./jdk-14.0.1/include/jni.h
./jdk-11.0.7/include/jni.h
./jdk-13.0.1/include/jni.h

$ find ./ -name jni_md.h
./jdk-14.0.1/include/linux/jni_md.h
./jdk-11.0.7/include/linux/jni_md.h
./jdk-13.0.1/include/linux/jni_md.h
```

Seems that chosen JDK's paths depend on order that they are found and processed. 

For example changing directory names from ```jdk-14.0.1``` to ```jdk-14.0.1_``` and from ```jdk-13.0.1``` to ```jdk-13.0.1_``` results in choosing paths going into Java 13.  
Changing name only of Java 14 SDK results in choosing paths going to Java 11 (as wanted).

**Comment by saudet** [comment]
And what does System.getProperty("java.home") return?

**Comment by krzysg** [comment]
Printing ```java.home``` property reveals the source of the problem and it is connected to searching ```jni.h``` in directory lying level up (via getParentFile()) to that returned by ```java.home```:

```java
javaHome = new File(System.getProperty("java.home")).getParentFile().getCanonicalFile();
 ```
In MacOS there is no problem since ```java.home``` is somewhere inside JDK root path so going level up in directory hierachy is OK and we stay within this particular JDK:

MacOS:
```
$ /Library/Java/JavaVirtualMachines/jdk1.8.0_151.jdk/Contents/Home/bin/java JavaHome 
/Library/Java/JavaVirtualMachines/jdk1.8.0_151.jdk/Contents/Home/jre

$ /Library/Java/JavaVirtualMachines/jdk-9.0.1.jdk/Contents/Home/bin/java JavaHome 
/Library/Java/JavaVirtualMachines/jdk-9.0.1.jdk/Contents/Home

$ /Library/Java/JavaVirtualMachines/jdk-14.0.1.jdk/Contents/Home/bin/java JavaHome 
/Library/Java/JavaVirtualMachines/jdk-14.0.1.jdk/Contents/Home
```

Same applies to older JDK on Linux and ```java.home``` there points to ```jre``` directory lying within particular JDK:

Linux:
```
$ /usr/lib/jvm/java-8-openjdk-amd64/bin/java JavaHome 
/usr/lib/jvm/java-8-openjdk-amd64/jre

$ /usr/lib/jvm/java-8-oracle/bin/java JavaHome 
/usr/lib/jvm/java-8-oracle/jre

$ /usr/lib/jvm/jdk1.8.0_152/bin/java JavaHome 
/usr/lib/jvm/jdk1.8.0_152/jre
```

And in above examples this is perfectly OK since ```include``` dir with ```jni.h``` is placed in a root dir of JDK so going 'level up' is a must.

However latest JDK from Oracle for Linux does not have ```jre``` directory and returned ```java.home``` is just a root dir for this JDK and by going 'level up' all installed JDKs are searched instead of one in use:

```
$ /usr/lib/jvm/jdk-14.0.1/bin/java JavaHome 
/usr/lib/jvm/jdk-14.0.1
```

Same applies to JDK 11 and 13 that I have installed.

So way to go would be first search ```java.home``` as it is and if not successful then going 'level up' in dir hierachy.

**Comment by saudet** [comment]
> So way to go would be first search `java.home` as it is and if not successful then going 'level up' in dir hierarchy.

Sounds like a good strategy. If you have some time, please send a pull request with a patch!

**Comment by saudet** [comment]
Fix released with JavaCPP 1.5.4! Thank you for your contribution
